### ISSUE TITLE
Add Certificate Serial Number & Certificate Attributes authn to Add-PASApplicationAuthenticationMethod

### ISSUE THREAD

**Issue by infamousjoeg** [pspete/psPAS#249]
**Is your feature request related to a problem? Please describe.**
When trying to use psPAS to automate the addition of a Certificate Serial Number or Certificate Attributes authentication for Client Certificate Authentication through CCP, `Add-PASApplicationAuthenticationMethods` does not have the capability to handle this.

**Describe the solution you'd like**
Additional AuthTypes `CertSerialNumber` and `CertAttributes` for the function.

**Describe alternatives you've considered**
Using `Invoke-RestMethod` cmdlet to handle it.

**Additional context**
N/A

**Comment by infamousjoeg** [comment]
Certificate Attributes can only be added via REST API: https://docs.cyberark.com/Product-Doc/OnlineHelp/PAS/Latest/en/Content/WebServices/Add%20Authentication.htm#certattrauthn

**Comment by pspete** [comment]
Thanks Joe - will lookinto this for the next update (coming soon).

Does this not work already?:

```powershell
Add-PASApplicationAuthenticationMethod -AppID SomeApp -AuthType certificateserialnumber -AuthValue 01NFAM0U50J0E0GARC1A00FA3DEFE9A9
```

**Comment by pspete** [comment]
@infamousjoeg - this feature is now present in the `dev` branch, so will be included in the next release.

This became a breaking change since I refactored the parameters; the new syntax for all supported operations of the the function will be:

```powershell
Add-PASApplicationAuthenticationMethod -AppID <String> [-Subject <String[]>] [-Issuer <String[]>] [-SubjectAlternativeName <String[]>] [-Comment <String>]

Add-PASApplicationAuthenticationMethod -AppID <String> -machineAddress <String>
    
Add-PASApplicationAuthenticationMethod -AppID <String> -osUser <String>
    
Add-PASApplicationAuthenticationMethod -AppID <String> -hash <String>
    
Add-PASApplicationAuthenticationMethod -AppID <String> -certificateserialnumber <String> [-Comment <String>]
    
Add-PASApplicationAuthenticationMethod -AppID <String> -path <String> [-IsFolder <Boolean>] [-AllowInternalScripts <Boolean>]
```

**Comment by pspete** [comment]
Now available in psPAS 3.5 - from the master branch, github release & PowerShell Gallery.
